server/src/assetComputeDevTool.js

```diff
@@ -258,7 +263,20 @@ async function setupCloudStorage() {
 /**
  * Setup the dev tool framework.
  */
+// singleton Promise
+let devToolPromise;
+
 async function setupAssetComputeDevTool() {
+    if (devToolPromise) {
+        return devToolPromise;
+    }
+    devToolPromise = new Promise((resolve) => {
+        resolve(doSetupAssetComputeDevTool());
+    });
+    return devToolPromise;
+}
+
+async function doSetupAssetComputeDevTool() {
```

Comment:
I would do the `checkJournal` polling inside the [`setUpAssetCompute()`](https://github.com/adobe/asset-compute-devtool/pull/22/files#diff-2be5cd0d59c0dfe0b6b9da45b8fa3a0aR230)...after the register call. That way the run is disabled until the journal has been fully set up.

The client side does not need to keep state about the journal readiness. It can all be handled on the server side.

Reply:
thanks, @jdelbick!
pls see comment above.. We have [another ticket](https://jira.corp.adobe.com/browse/NUI-667) to redesign this behavior a bit, so may be can handle it there and remove this altogether then.